source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -272,111 +272,211 @@ absl::Status ProcessorState::handleHeadersResponse(const HeadersResponse& respon
   return absl::FailedPreconditionError("spurious message");
 }
 
-// TODO(#37048) Refactoring this function by adding one helper function for each callback state.
+/**
+ * Handles responses containing body modifications from an external processor. Supports three modes
+ * of operation: buffered, streamed, and buffered partial.
+ *
+ * @param response The body response received from the external processor
+ * @return Status indicating success or failure of the handling operation
+ */
```

Comment:
@agrawroh Thanks for refactoring! 

Just the style nit from a quick drive-by : They are nice comments but i think it is better to be added to the function declaration in the header file